packages/core/src/auto.ts

```diff
@@ -1760,8 +1775,12 @@ export default class Auto {
       dryRun: options.dryRun,
       quiet: options.quiet,
     });
+
+    // New new version must be computed, so make release context
+    const releaseArgs = await this.makeReleaseContext();
+
     this.logger.verbose.info("Calling after version hook");
-    await this.hooks.afterVersion.promise({ dryRun: options.dryRun });
+    await this.hooks.afterVersion.promise({ dryRun: options.dryRun, version: releaseArgs.newVersion });
```

Comment:
OH I see why you do all of this rearranging. in the afterVersion hook if it's not a dry run it IS a release. We don't have to care about the return value of makeRelease. The undefined that it returns is for it's other usage. Here we know a release (and thus a version) has happened. 

You can revert all the changes to core and remove references in you plugin to version and it still all works!